### PR TITLE
Vaults to spaces - move permissions to ACLs

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

As we move from vaults to spaces, we want to leverage multiple groups associated with one vault. The current implementation applies permissions checks directly on the vault, ignoring group-specific nuances. This PR shifts the logic to ACLs, tailoring permissions to user roles as our primary source of truth for now (see more details [here](https://github.com/dust-tt/dust/pull/8091)).

Main changes:
- simplified `canWrite` and `canRead` methods, now consistently using `auth.canRead` and `auth.canWrite`
- centralized vault read and write logic for each group kind in one place
- improved flexibility for future granular permission adjustments.

The logic is implemented in `VaultResource` rather than `GroupResource` for two main reasons:
- it keeps permission logic close to vault functionality
- it allows for future granular permission (that customers already start asking around about) (e.g., non-admin/builder users that can add members to a vault, ...)

This approach sets the foundation for more sophisticated permission models, such as storing specific permissions in the `group_vault` join table, enabling fine-grained access control per group within each vault/space.

## Risk

Pretty risky, as permissions are the gatekeeper to our data. I tested all three roles, run queries on private API using all three roles. Safe to rollback.

### CODE REVIEW

**Review by spolu — COMMENTED** [review]
Left a question 

But also I think it's a good time to clean up the can admin concept.

Generally looks like the right thing to do otherwise 👍👍

File: front/lib/resources/vault_resource.ts [hunk 1]
```diff
@@ -325,11 +326,39 @@ export class VaultResource extends BaseResource<VaultModel> {
...
+  ): Permission[] {
+    switch (group.kind) {
+      case "global":
+        return auth.isBuilder() ? ["read", "write"] : ["read"];
+
+      case "regular":
+        return ["read", "write"];
+
+      case "system":
+        return auth.isAdmin() ? ["read", "write"] : ["read"];
```
Comment (spolu) [inline front/lib/resources/vault_resource.ts:350]: Can we really read the system group? Surely it's not true?

**Review by Fraggle — COMMENTED** [review]
As we discussed, I still find the way permissions are handled to be convoluted.

We think it's the ACL but actually the truth is split:
- in the `getGroupPermissionsForVault`, we bypass based on the role if "isBuilder"
- we have canRead, canWrite etc.. where we also potentially bypass based on role (for admins in particular).

(i agree with @spolu for the addition of "admin" permission)

Taking a step back, the permissions on a resource are based on the role or the groups of the user.

Therefor, I suggest evolving the `ACLType` from:

```
[
 {
    groupId: group.id,
    permissions: ["read", "write"]
 },...
]
```

to 

```
{
   groups: [
    {
     id: group.id,
     permissions: ["read", "write", "admin"]
    },...
   ],
   roles: [{
     name: "admin" | "builder" | "user",
     permissions: ["read", "write", "admin"]
    },...
   ]
 }
```

The vault acl() method returns the right ACLs based on the vault kind.

It would then be a simple lookup in the ACLs to determine the permissions of an user and the truth would be in a single place.

**Comment by Fraggle** [comment]
@flvndvd we could refine it further to have group "kind" permissions in the ACL.

group kind, group id, role

**Comment by spolu** [comment]
What's the status here? I thought this PR was blocking the one referred to in the PR description?

**Comment by flvndvd** [comment]
> What's the status here? I thought this PR was blocking the one referred to in the PR description?

I figured this might bring some discussions, so I used a temporary workaround to keep the transition to Spaces on track. I'll get back to this tomorrow.

**Comment by flvndvd** [comment]
Superseeded by https://github.com/dust-tt/dust/pull/8209.

**Pull request closed without merging**